harvester_e2e_tests/integration/test_vm.py

```diff
@@ -0,0 +1,295 @@
+    "harvester_e2e_tests.fixtures.api_client"
+]
+
+
+@pytest.fixture(scope="session")
+def focal_image_url():
+    return "http://10.84.128.164:8081/harvester_vm_images_for_tests/focal-server-cloudimg-amd64.img"  # noqa
+
+
+@pytest.fixture(scope="class")
+def focal_image(api_client, unique_name, focal_image_url, wait_timeout):
+    code, data = api_client.images.create_by_url(unique_name, focal_image_url)
+    assert 201 == code, (
+        f"Failed to upload focal image with error: {code}, {data}"
+    )
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.images.get(unique_name)
+        if 'status' in data and 'progress' in data['status'] and \
+                data['status']['progress'] == 100:
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Image {unique_name} can't be ready with {wait_timeout} timed out\n"
+            f"Got error: {code}, {data}"
+        )
+
+    namespace = data['metadata']['namespace']
+    name = data['metadata']['name']
+
+    yield dict(ssh_user="ubuntu", id=f"{namespace}/{name}")
+
+    api_client.images.delete(name, namespace)
+
+
+@pytest.fixture(scope="class")
+def available_node_names(api_client):
+    status_code, nodes_info = api_client.hosts.get()
+    assert status_code == 200, f"Failed to list nodes with error: {nodes_info}"
+
+    node_names = []
+    for node_info in nodes_info.get('data', []):
+        is_ready = False
+        for condition in node_info.get('status', {}).get('conditions', []):
+            if condition.get('type', "") == "Ready" and \
+                    condition.get('status', "") == "True":
+                is_ready = True
+                break
+
+        if is_ready and not node_info.get('spec', {}).get('unschedulable', False):
+            node_names.append(node_info['metadata']['name'])
+
+    assert 2 <= len(node_names), (
+        f"The cluster only have {len(node_names)} available node. It's not enough."
+    )
+    yield node_names
+
+
+def wait_vm_agent_connected(api_client, vm_name, wait_timeout):
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    code, data = None, None
+    while endtime > datetime.now():
+        code, data = api_client.vms.get(vm_name)
+        if data.get('status', {}).get('ready', False):
+            code, data = api_client.vms.get_status(vm_name)
+            if data['status']['conditions'][-1]['status'] == 'True':
+                break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Can't find VM {vm_name} with {wait_timeout} timed out\n"
+            f"Got error: {code}, {data}"
+        )
+
+
+@pytest.mark.multivmtest
+@pytest.mark.virtual_machines_p1
+def test_multiple_migrations(api_client, unique_name, focal_image, wait_timeout,
+                             available_node_names):
+    first_vm_name = f"{unique_name}-1"
+    second_vm_name = f"{unique_name}-2"
```

Comment:
I will suggest use
```python
vms = [f"migrate-1-{unique_name}", f"migrate-2-{unique_name}"]
```

The unique_name would be very long, so add prefix would better than suffix (when the name is meaningful)